### ISSUE TITLE
Use generators instead of creating list.

### ISSUE THREAD

**Issue by tshirtman** [wigging/advent-of-code#1]
your get_lines() function reads through the input file, and accumulate the lines in a list using append, this is not great for multiple reasons.

- Ultimately the whole file needs to fit in memory, it's fine with simple problems like this, but if you have to process a large file, you might explode your memory for no useful reason.
- Python lists are nice and never ask you how much items you will need to fit in them, but they *have* to decide for some space, usually a relatively low value, and when the list is full, python will transparently create a new list with more space and copy over all the items in the old list to the new one, it's thankfully smart and amortize this cost by increasing the amount or reserved memory proportionally to the current size, but still, if you create a very large list this way, you might copy things many time as the list grows.

Instead, python has `generators`, when you do.
`for line in file:` file is not a list, it's a generator, and you can do one thing with each line and when you iterate, the memory for the previous line can be freed, so you never have all the lines in memory. Since you wrapped this in a get_lines() function to use rstrip() automatically, you need to make that function a generator too. the solution to do that is simple, use "yield line.rstrip()" in your loop, yield works a bit like return, except the function can be continued later, working in step with the for loop that calls it.

```py
def get_lines(filename):
    with open(filename) as file:
         for line in file:
            yield line.rstrip()
```

```py
for line in get_lines("input_day1"):
    ...
```

**Comment by wigging** [comment]
Thank you for the suggestion. When you do `for line in get_lines("input_day1"):` this opens the file on each iteration. Seems like it would be better to open the file once instead of multiple times.

**Comment by tshirtman** [comment]
no, this doesn't open the file multiple times, get_lines() returns an iterator, every time go go through the loop, next() is called on the iterator, resuming the function (not starting over!) until it hits the next yield.

**Comment by wigging** [comment]
Ok, that makes sense. Thanks again for pointing this out. I made the changes to my code.
